File: packages/mobile/lib/features/pokemon_list/domain/entities/pokemon_entity.dart
abstract class PokemonEntity {
  final int id;
  final String name;
  final List<String> types;
  final List<String> abilities;
  final double height;
  final double weight;
  final double baseExperience;
  final String imageUrl;

  PokemonEntity({
    required this.id,
    required this.name,
    required this.types,
    required this.abilities,
    required this.height,
    required this.weight,
    required this.baseExperience,
    required this.imageUrl,
  });
}

File: bff/src/domain/entities/pokemon.entity.ts
export interface Pokemon {
  id: number;
  name: string;
  baseExperience: number;
  height: number;
  weight: number;
  sprites: {
    frontDefault: string;
    backDefault: string;
    frontShiny: string;
    backShiny: string;
  };
  types: {
    slot: number;
    type: {
      name: string;
      url: string;
    };
  }[];
  abilities: {
    ability: {
      name: string;
      url: string;
    };
    isHidden: boolean;
    slot: number;
  }[];
  stats: {
    baseStat: number;
    effort: number;
    stat: {
      name: string;
      url: string;
    };
  }[];
  moves: {
    move: {
      name: string;
      url: string;
    };
    versionGroupDetails: {
      levelLearnedAt: number;
      moveLearnMethod: {
        name: string;
        url: string;
      };
      versionGroup: {
        name: string;
        url: string;
      };
    }[];
  }[];
  species: {
    name: string;
    url: string;
  };
  evolutionChain?: {
    url: string;
  };
}
